File: Dsa-day64.py
class Spreadsheet:
    def __init__(self, rows: int):
        self.cells = {}

    def setCell(self, cell: str, value: int) -> None:
        self.cells[cell] = value

    def resetCell(self, cell: str) -> None:
        if cell in self.cells:
            del self.cells[cell]

    def getValue(self, formula: str) -> int:
        def get_val(x):
            if x[0].isalpha():
                return self.cells.get(x, 0)
            return int(x)
        parts = formula[1:].split('+')
        return get_val(parts[0]) + get_val(parts[1])

File: Dsa-day64.java
import java.util.*;
class Spreadsheet {
    private Map<String,Integer> cells;
    public Spreadsheet(int rows) {
        cells=new HashMap<>();
    }
    public void setCell(String cell,int value) {
        cells.put(cell,value);
    }
    public void resetCell(String cell) {
        cells.remove(cell);
    }
    public int getValue(String formula) {
        String[] parts=formula.substring(1).split("\\+");
        return getVal(parts[0])+getVal(parts[1]);
    }
    private int getVal(String s) {
        if(Character.isLetter(s.charAt(0))) {
            return cells.getOrDefault(s,0);
        }
        return Integer.parseInt(s);
    }
}
